### PR TITLE
Remove the phonebook from the node

### PR DESCRIPTION
**Pull request by algonautshant**
## Summary
The phonebook object is removed from the node. The phonebook is now local to the network. 
The node only passes list of addresses in string type to the network upon creation. 

## Test Plan

The node_test.go has many skipped test points relevant to this job. 
https://github.com/algorand/go-algorand/issues/892 is tracking this.

### CODE REVIEW

File: cmd/algod/main.go [hunk 1]
```diff
@@ -249,7 +249,23 @@ func main() {
 		log.Fatalf("DefaultDeadlock is somehow not set to an expected value (enable / disable): %s", config.DefaultDeadlock)
 	}
 
-	err = s.Initialize(cfg)
+	var phonebookAddressees []string
```
Comment (tsachiherman) [inline cmd/algod/main.go]: typo `phonebookAddressees` -> `phonebookAddresses`

File: agreement/gossip/networkFull_test.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func spinNetwork(t *testing.T, nodesCount int) ([]*networkImpl, []*messageCounte
 	for nodeIdx := 0; nodeIdx < nodesCount; nodeIdx++ {
 		phonebooks[nodeIdx] = network.MakePhonebook(cfg.ConnectionsRateLimitingCount,
 			time.Duration(cfg.ConnectionsRateLimitingWindowSeconds)*time.Second)
-		gossipNode, err := network.NewWebsocketGossipNode(log.With("node", nodeIdx), cfg, phonebooks[nodeIdx], "go-test-agreement-network-genesis", config.Devtestnet)
+		gossipNode, err := network.NewWebsocketGossipNode(log.With("node", nodeIdx), cfg, nodesAddresses, "go-test-agreement-network-genesis", config.Devtestnet)
```
Comment (tsachiherman) [inline agreement/gossip/networkFull_test.go]: I think that you should be able to delete the `phonebooks := make([]network.Phonebook, nodesCount)` line.

File: cmd/algod/main.go [hunk 3]
```diff
@@ -249,7 +249,23 @@ func main() {
...
+	if peerOverrideArray != nil {
+		phonebookAddressees = peerOverrideArray
+	} else {
+		ex, err := os.Executable()
+		if err != nil {
+			log.Errorf("cannot locate node executable: %s", err)
```
Comment (tsachiherman) [inline cmd/algod/main.go]: if you hit this line you shouldn't keep going with `phonebookDir := filepath.Dir(ex)`, right ?

File: node/node.go [hunk 4]
```diff
@@ -140,25 +139,17 @@ type TxnWithStatus struct {
 
 // MakeFull sets up an Algorand full node
 // (i.e., it returns a node that participates in consensus)
-func MakeFull(log logging.Logger, rootDir string, cfg config.Local, phonebookDir string, genesis bookkeeping.Genesis) (*AlgorandFullNode, error) {
+func MakeFull(log logging.Logger, rootDir string, cfg config.Local, phonebookAddressees []string, genesis bookkeeping.Genesis) (*AlgorandFullNode, error) {
```
Comment (tsachiherman) [inline node/node.go]: typo `phonebookAddressees`

File: cmd/algod/main.go (hunk 3 above)
Comment (algonautshant) [reply cmd/algod/main.go]: yes. you're right.

File: agreement/gossip/networkFull_test.go (hunk 2 above)
Comment (algonautshant) [reply agreement/gossip/networkFull_test.go]: yes.

**Pull request merged**